### PR TITLE
Hex drawing

### PR DESCRIPTION
**Pull request by Chroq**
## Description

In this PR, I'm drawing the board and place some stones on cell.

Trello : [https://trello.com/c/HYLkatPg/5-as-norbert-i-want-to-be-able-to-drop-some-stones-on-a-board](url)

## Expectations

Screenshot_20191210-142901_Expo

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
-import {StyleSheet, Text, View} from 'react-native';
+import {StyleSheet, Text, View, TouchableOpacity, Button} from 'react-native';
+import Svg, {Polygon, SvgXml} from 'react-native-svg';
```
Comment (alexisjanvier) [inline App.js]: :+1:

File: App.js [hunk 2]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
...
+                <Svg height="100" width="100">
+                    <SvgXml xml={xml} width="100%" height="100%"/>
+                </Svg>
+            </TouchableOpacity>
+        );
+    }
+}
+
+
+class Board extends React.Component {
```
Comment (alexisjanvier) [inline App.js]: IMO, The board should be isolated in a `Board.js` file

File: App.js [hunk 3]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
...
+const xml = `<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="174" height="200"
+                 viewBox="0 0 173.20508075688772 200">
+                <path fill="#EEEEEE" d="M86.60254037844386 0L173.20508075688772 50L173.20508075688772 150L86.60254037844386 200L0 150L0 50Z"></path>
+            </svg>`
+
+class Stone extends React.Component {
+    render() {
+        return (
+            <TouchableOpacity onPress={this._onPressButton}>
+                <Svg height="100" width="100">
```
Comment (alexisjanvier) [inline App.js]: I'm not sure if it's necessary to return each cell (it's more of a `<Cell />` than a `<Stone />`, by the way) in a `<Svg />` component. It is probably the complete board that should be.

File: App.js [hunk 4]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
...
+                    <SvgXml xml={xml} width="100%" height="100%"/>
+                </Svg>
+            </TouchableOpacity>
+        );
+    }
+}
+
+
+class Board extends React.Component {
+    renderStone(i) {
```
Comment (alexisjanvier) [inline App.js]: No need for this method of rendering. Use directly the component `<Stone />` in the returned jsx !

File: App.js [hunk 5]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
...
+        return (
+            <View>
+                <Text className="status">{status}</Text>
+                <View style={{flex: 1, flexDirection: 'row'}} className="board-row">
+                    {this.renderStone(0)}
+                    {this.renderStone(1)}
+                    {this.renderStone(2)}
+                </View>
+                <View style={{flex: 1, flexDirection: 'row'}} className="board-row">
+                    {this.renderStone(3)}
```
Comment (alexisjanvier) [inline App.js]: {this.renderStone(3)} => <Stone /> 
Was that the cause of your reported error in Slack?

File: package.json [hunk 6]
```diff
@@ -16,7 +16,8 @@
     "react": "16.8.3",
     "react-dom": "16.8.3",
     "react-native": "https://github.com/expo/react-native/archive/sdk-35.0.0.tar.gz",
-    "react-native-web": "^0.11.7"
+    "react-native-web": "^0.11.7",
```
Comment (alexisjanvier) [inline package.json]: It is a good habit to be consistent on dependency version management. IMO you have to choose the fixed version or the `^`, but not both...

File: App.js [hunk 7]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
...
+                    {this.renderStone(6)}
+                    {this.renderStone(7)}
+                    {this.renderStone(8)}
+                </View>
+            </View>
+        );
+    }
+}
+
+// class Game extends React.Component {
```
Comment (alexisjanvier) [inline App.js]: To be deleted if no longer used

File: package.json (hunk 6 above)
Comment (fzaninotto) [reply package.json]: Choose the ^ version, it relaxes version constraints and allows npm/yarn to avoid duplicates

File: App.js [hunk 8]
```diff
@@ -17,3 +19,64 @@ const styles = StyleSheet.create({
         justifyContent: 'center',
     },
 });
+
+const xml = `<svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="174" height="200"
+                 viewBox="0 0 173.20508075688772 200">
+                <path fill="#EEEEEE" d="M86.60254037844386 0L173.20508075688772 50L173.20508075688772 150L86.60254037844386 200L0 150L0 50Z"></path>
+            </svg>`
+
+class Stone extends React.Component {
```
Comment (fzaninotto) [inline App.js]: this one should live in another file. Also, prefer functional components over class components (less boilerplate)

File: package.json (hunk 6 above)
Comment (Kmaschta) [reply package.json]: Also, `package-lock.json` and `yarn.lock` are freezing the dependencies, having `^` allows to upgrade depencencies quickly.

Comment (Chroq) [reply package.json]: Ok, I have tried to change with the "^" version but it's bring a lot of sides effect... I'll try to change that later and I have fixed all dependencies.

File: board/Cell.js [hunk 9]
```diff
@@ -2,12 +2,10 @@ import React from 'react';
 import {corners} from './utils';
 import {Polygon} from "react-native-svg";
 
-const Cell = (props) => {
+function Cell(props) {
```
Comment (zyhou) [inline board/Cell.js]: `const Cell = (props) => {` and `function Cell(props) {` are equal. You can keep the first style

File: README.md [hunk 10]
```diff
@@ -21,4 +21,4 @@ make test
 ```
 ## License
 
-Hex React is licensed under the [MIT License](LICENSE), courtesy of [Marmelab](http://marmelab.com).
+Cell React is licensed under the [MIT License](LICENSE), courtesy of [Marmelab](http://marmelab.com).
```
Comment (alexisjanvier) [inline README.md]: Why did you change the project name ?

File: board/utils.js [hunk 11]
```diff
@@ -0,0 +1,92 @@
+/* eslint-disable no-mixed-operators,consistent-return */
```
Comment (alexisjanvier) [inline board/utils.js]: You can put the link to the original lib ;)

File: board/utils.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+        const rad = Math.PI / 180 * (baseDeg + diff);
+        return [x + size * Math.cos(rad), y + size * Math.sin(rad)];
+    });
+};
+
+export const gridPoint = (oType, oX, oY, size, gridX, gridY) => {
+    /* eslint-disable no-else-return */
+    /* reason: it seems buggy */
+    if (oType === POINTY) {
+        const height = size * 1;
```
Comment (alexisjanvier) [inline board/utils.js]: `size * 1` ^^

File: board/utils.js [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+                type: oType,
+                x,
+                y,
+                size,
+            },
+            gridX,
+            gridY,
+            corners: hexCorners(oType, x, y, size),
+        };
+    } else if (oType === FLAT) {
```
Comment (alexisjanvier) [inline board/utils.js]: Do you plan to use type `FLAT`? If not, you can delete the condition

File: board/index.js [hunk 14]
```diff
@@ -1,13 +1,15 @@
...
 
 import {Cell} from "./Cell";
 import {gridPoints} from "./utils"
 
 export const Board = (props) => {
+
     const grid = gridPoints(50, 50, 22, props.size, props.size).map(({props}, index) => (
-        <Cell {...props} onPress={() => {
-        }} fill="#FFFFFF" stroke="#222222" key={index}/>
+        <Cell onPress={() => {
```
Comment (alexisjanvier) [inline board/index.js]: Your state is managed by the `Cell` component. So you will have to bind the change of this state within this component (on `<Polygon />`).

Comment (alexisjanvier) [reply board/index.js]: Moreover, you want to manage the color of a cell (with a fill and not a color props!). You no longer need to pass the `fill` props to the`<Cell />` component. 
It is Cell's state that will control this fill value assigned to the `<Polygon />`

File: board/Cell.js [hunk 15]
```diff
@@ -1,9 +1,12 @@
...
 import {Polygon} from "react-native-svg";
 
 export const Cell = (props) => {
     const {x, y, size} = props;
     const points = corners(x, y, size);
-    return <Polygon  {...props} points={points.map(p => p.join(',')).join(' ')}/>;
+
+    const [color, setColor] = useState("white");
+
+    return <Polygon {...props} color={color} points={points.map(p => p.join(',')).join(' ')}/>;
```
Comment (alexisjanvier) [inline board/Cell.js]: Here, you can make the fill value depend on the state. 
And it is also here that you can associate the `onPress` event with the function to change the state.

File: board/Cell.js [hunk 16]
```diff
@@ -1,9 +1,12 @@
-import React from 'react';
+import React, {useState, useEffect} from "react";
```
Comment (alexisjanvier) [inline board/Cell.js]: :+1:  Good idea to use hooks

File: board/Cell.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+import {Polygon} from "react-native-svg";
+
+export const Cell = (props) => {
+    const {x, y, size} = props;
+    const points = corners(x, y, size);
+
+    const [color, setColor] = useState("white");
+
+    return <Polygon onPress={() => {
+        setColor("#03a9fc")
```
Comment (alexisjanvier) [inline board/Cell.js]: It is a good practice to isolate this in a function like `handlePlay`... But okay for now.

File: board/utils.js [hunk 18]
```diff
@@ -0,0 +1,68 @@
+/*
+ * react-hex
+ *
+ * Credits : https://github.com/airtoxin/react-hex
```
Comment (Luwangel) [inline board/utils.js]: You couldn't import this lib with npm instead of copying the source code?

Comment (alexisjanvier) [reply board/utils.js]: @Luwangel Nope, this lib doesn't work on react native ....

File: board/index.js [hunk 19]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {Polygon, Svg} from 'react-native-svg';
+
+import {Cell} from "./Cell";
+import {gridPoints} from "./utils"
+
+export const Board = (props) => {
+
+    const grid = gridPoints(50, 50, 22, props.size, props.size).map(({props}, index) => (
+        <Cell {...props} stroke="#222222" key={index}/>
```
Comment (fzaninotto) [inline board/index.js]: I find it strange that you duplicate the props to each cell. Props forwarding usually concerns only one component.

File: board/index.js [hunk 20]
```diff
@@ -0,0 +1,18 @@
+import React from "react";
+import {Polygon, Svg} from 'react-native-svg';
+
+import {Cell} from "./Cell";
+import {gridPoints} from "./utils"
+
+export const Board = (props) => {
+
+    const grid = gridPoints(50, 50, 22, props.size, props.size).map(({props}, index) => (
```
Comment (fzaninotto) [inline board/index.js]: I don't really see the point to extract that as a constant. In React, it's common to inline this type of iteration inside the JSX.

File: board/Cell.js [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+    }} {...props} fill={color} points={points.map(p => p.join(',')).join(' ')}/>;
```
Comment (fzaninotto) [inline board/Cell.js]: You shouldn't pass all the props. `x`, `y`, and `size` concern `Cell` and not `Polygon`.

File: board/index.js (hunk 19 above)
Comment (Chroq) [reply board/index.js]: Each props are different, no ? It's the map who generate it so they look like (for the last cell) :
```
Object {
  "number": 35,
  "onPress": [Function onPress],
  "player": 0,
  "size": 22,
  "x": 160.25,
  "y": 157.5,
}
``` 
However, I must concede that building process isn't the clearest.

File: board/index.js (hunk 20 above)
Comment (Chroq) [reply board/index.js]: Changed ! TY

File: board/Cell.js (hunk 21 above)
Comment (Chroq) [reply board/Cell.js]: I have turn this review into an issue. TY

File: board/index.js (hunk 19 above)
Comment (fzaninotto) [reply board/index.js]: You're right. I think my confusion comes from the fact that you're using `props` twice, with two different values. That's not very readable.
